cpu/esp32/esp-wifi/doc.txt

```diff
@@ -7,12 +7,42 @@
  */
 
 /**
- * @defgroup        cpu_esp32_esp_wifi ESP WiFi netdev interface
+ * @defgroup        cpu_esp32_esp_wifi ESP32 WiFi netdev interface
  * @ingroup         cpu_esp32
- * @brief           WiFi AP-based network device driver
- *
- * This module realizes a netdev interface using the built-in
- * WiFi module and AP infrastructure.
+ * @brief           Network device driver for the ESP32 WiFi interface
  *
  * @author      Gunar Schorcht <gunar@schorcht.net>
+
+This module realizes a `netdev` interface for the built-in WiFi interface
+of ESP32. To enable the WiFi interface, module `esp_wifi` has to be used.
+
+@note Due to symbol conflicts with the `crypto` and `hash` modules of RIOT
```

Comment:
side note: we should look into that, could easily get annoying.

Reply:
The problem is that `wpa_supplicant` used by module `esp_wifi` includes its own crypto library. This crypto library defines the same functions as module `crypto`, but unfortunately also some more that are not provided by module `crypto`.

Reply:
Missing function are:
```
aes_wrap
aes_encrypt_init
aes_encrypt_deinit
aes_decrypt_init
aes_decrypt_deinit
```
I have tried to use the `crypto` module instead of the `wpa_supplicant` crypto library, ESP32 cannot connect the AP anymore due to an authentication problem.

Reply:
ah so you think if we provide/implement these for the RIOT crypto modules, you can replace the ESP stuff with those? If so, it might be good if you open an issue on that matter, to keep track of this.

Reply:
> ah so you think if we provide/implement these for the RIOT crypto modules, you can replace the ESP stuff with those?

Maybe. For example, it seems that `aes_encrypt_init` seems to do something similar to `aes_init`. The difference is that `aes_encrypt_init` returns a pointer to the cipher context while `aes_init` returns an int. A wrapper arround `aes_init` might help. I'm already trying something.

Reply:
I opened an issue #10787